src/jit/lsra.cpp

```diff
@@ -4967,27 +5061,39 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
                     RefPosition* nextAssignedRef = recentAssignedRef->nextRefPosition;
                     assert(nextAssignedRef != nullptr);
                     assert(nextAssignedRef->nodeLocation == refLocation ||
-                           (nextAssignedRef->nodeLocation + 1 == refLocation && nextAssignedRef->delayRegFree));
+                        (nextAssignedRef->nodeLocation + 1 == refLocation && nextAssignedRef->delayRegFree));
                 }
             }
             continue;
         }
-            
+
         // If we have a recentAssignedRef, check that it is going to be OK to spill it
+        //
+        // TODO-Review: Under what conditions recentAssginedRef would be null?
+        unsigned recentAssignedRefWeight = BB_ZERO_WEIGHT;
         if (recentAssignedRef != nullptr)
         {
             if (recentAssignedRef->nodeLocation == refLocation)
             {
                 // We can't spill a register that's being used at the current location
-                RefPosition * physRegRef = physRegRecord->recentRefPosition;
+                RefPosition* physRegRef = physRegRecord->recentRefPosition;
                 continue;
             }
-        
+
             // If the current position has the candidate register marked to be delayed, 
             // check if the previous location is using this register, if that's the case we have to skip
             // since we can't spill this register.
-            if(recentAssignedRef->delayRegFree &&
-               (refLocation == recentAssignedRef->nodeLocation + 1))
+            if (recentAssignedRef->delayRegFree &&
+                (refLocation == recentAssignedRef->nodeLocation + 1))
+            {
+                continue;
+            }
+
+            // We don't prefer to spill a register if the weight of recentAssignedRef > weight
+            // of the spill candidate found so far.  We would consider spilling a greater weight
+            // ref position only if the refPosition being allocated must need a reg.
+            recentAssignedRefWeight = getWeight(recentAssignedRef);
```

Comment:
This will cause the weight of the RefPosition to be computed every time it is compared against another RefPosition, even though it is the same weight for all RefPositions of a given Interval. I wonder if there is a measurable throughput impact to doing so. If so, one might consider just computing the weight of the current RefPosition, and then when it is assigned to a register, keep the most recent assigned weight in the associated RegRecord. That would avoid some of the recomputation, without having to add it to each Interval (of which there are generally more than there are RegRecords).